### PR TITLE
Skip unittests that request it rather than mark them as failed. Close…

### PR DESCRIPTION
**Pull request by jpakkane**
…s #1804.

### CODE REVIEW

**Comment by DimStar77** [comment]
Tested the patch on the openSUSE build system - the test suite of 0.40.1 successfully passed on:

* i586
* x86_64
* armv7
* aarch64
* ppc64le

Thanks for the fix

File: run_unittests.py [hunk 1]
```diff
@@ -381,6 +381,8 @@ def _run(self, command, workdir=None):
         output = p.communicate()[0]
         print(output)
         if p.returncode != 0:
+            if 'MESON_SKIP_TEST' in output:
+                raise unittest.SkipTest('Project requested skipping.')
```
Comment (nirbheek) [inline run_unittests.py]: Nitpick: "Project test case" instead?

Also this doesn't tell the reader what test case caused the skip.

Comment (QuLogic) [inline run_unittests.py]: Doesn't need to; `unittest` keeps track of the test cases.

Comment (nirbheek) [inline run_unittests.py]: I'm not talking about the unit test, I'm talking about the specific project test that requested a skip. You could look at the unit test code and see what project test it was using, but that is not straightforward for a lay user, and a unit test may use more than one project test anyway.

**Comment by jpakkane** [comment]
Now prints the name of the test causing the skip.

**Pull request merged**